play/src/front/Phaser/Game/MapEditor/AreasPropertiesListener.ts

```diff
@@ -203,6 +207,11 @@ export class AreasPropertiesListener {
 
                 break;
             }
+            case "matrixRoomPropertyData": {
+                this.handleMatrixRoomAreaOnEnter(property);
+                break;
+            }
+
```

Comment:
The `addPropertyFilter`, `updatePropertyFilter` and `removePropertyFilter` are synchronous.
As a result, any async operation might cause a race condition.

For instance, if we enter/leave an area very quickly, the code for the leave could finish before the code for enter.

Ideally, we should:

- return a Cancelable promise.
- keep track of Cancelable promise in a map per "type"
- -
- call the cancel promise of the "enter" if we go in a "leave" state (and vice versa?)